### PR TITLE
Alexey/default/values

### LINKED ISSUE #1265: yaml: deny section does not respect rule conditions
When configuring YAML, deny section should respect rule conditions

### PR DESCRIPTION
**Pull request by alex-kovoy**
if V3 spec `deny` section has empty namespace value, populate it with `default` namespace instead.

closes #1265

### CODE REVIEW

File: lib/services/resource.go [hunk 1]
```diff
@@ -335,11 +335,8 @@ func (m *Metadata) SetTTL(clock clockwork.Clock, ttl time.Duration) {
 	m.Expires = &expireTime
 }
 
-// CheckAndSetDefaults checks validity of all parameters and sets defaults
-func (m *Metadata) CheckAndSetDefaults() error {
-	if m.Name == "" {
-		return trace.BadParameter("missing parameter Name")
-	}
+// SetDefaults sets default values if missing
+func (m *Metadata) SetDefaults() {
```
Comment (klizhentas) [inline lib/services/resource.go]: this is a common method name, so can you bring it back as well as it's return value

Comment (klizhentas) [reply lib/services/resource.go]: ah, misread that

Comment (klizhentas) [reply lib/services/resource.go]: actually, let's not complicate things - let's have only one method `CheckAndSetDefaults` that does both things for now.

Comment (alex-kovoy) [reply lib/services/resource.go]: This is a `new` method used to SetDefaults and returns no errors as opposed to CheckAndSetDefaults. This method is then used in a few other places, like `Equals`

**Review by klizhentas — CHANGES_REQUESTED** [review]
want to make some changes to the methods signatures

File: lib/services/migrations_test.go [hunk 2]
```diff
@@ -284,6 +284,9 @@ func (s *MigrationsSuite) TestMigrateRoles(c *C) {
 					NewRule(KindRole, RW()),
 				},
 			},
+			Deny: RoleConditions{
+				Namespaces: []string{"default"},
```
Comment (klizhentas) [inline lib/services/migrations_test.go]: use constant instead of hardcoded value here and below

File: lib/services/resource.go (hunk 1 above)
Comment (alex-kovoy) [reply lib/services/resource.go]: It's also used in V2 -> V3 converter `func (r *RoleV2) V3() *RoleV3 `

File: lib/services/role.go [hunk 3]
```diff
@@ -493,6 +489,25 @@ func (r *RoleV3) CheckAndSetDefaults() error {
 	if r.Spec.Allow.Rules == nil {
 		r.Spec.Allow.Rules = CopyRulesSlice(AdminUserRules)
 	}
+	if r.Spec.Deny.Namespaces == nil {
+		r.Spec.Deny.Namespaces = []string{defaults.Namespace}
+	}
+
+	// check and correct the session ttl
+	maxSessionTTL, ok := r.Spec.Options.GetDuration(MaxSessionTTL)
```
Comment (kontsevoy) [inline lib/services/role.go]: `ok` needs to be renamed to `err`. Golang has an idiom (getting value from a map) where `ok` as a 2nd parameter is a boolean, indicating that a value is found or not.

**Review by kontsevoy — CHANGES_REQUESTED** [review]

**Pull request closed without merging**